src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/KeyValuePairModelBinder.cs

```diff
@@ -17,9 +17,13 @@ public sealed class KeyValuePairModelBinder<TKey, TValue> : IModelBinder
                 throw new ArgumentNullException(nameof(bindingContext));
             }
 
-            ModelBindingHelper.ValidateBindingContext(bindingContext,
-                                                      typeof(KeyValuePair<TKey, TValue>),
-                                                      allowNullModel: true);
+            if (bindingContext.ModelType != typeof(KeyValuePair<TKey, TValue>))
+            {
+                var message = Resources.FormatModelBinderUtil_ModelTypeIsWrong(
+                    bindingContext.ModelType,
+                    typeof(KeyValuePair<TKey, TValue>));
+                throw new ArgumentException(message, nameof(bindingContext));
+            }
```

Comment:
Would it be all that harmful to just have this return instead of throwing? That's more inline with how model binders actually work. Then we could kill the badly named resource.